### PR TITLE
[dynamo] Simplify `ConstantVariable.create` and` ConstantVariable.__init__`

### LINKED ISSUE #110871: [dynamo] Add asserts to prevent user defined objects/classes from going into ConstantVariable
### 🐛 Describe the bug

This is the root cause of https://github.com/pytorch/pytorch/pull/110794 but it's a bit hard to educate everyone not to put random crap in ConstantVariable, so it would be good if we have more asserts that help people figure out.

Primary barrier to adding the assert is we'll probably have to fix a pile of latent bugs while we're at it.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @Xia-Weiwen @aakhundov @Fidget-Spinner 

### Versions

main

### PR DESCRIPTION
**Pull request by StrongerXi**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #140984
* __->__ #140745

This patch removes some redundant code paths in
`ConstantVariable.create` and` ConstantVariable.__init__`.

Closes #110871.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

### CODE REVIEW

**Comment by StrongerXi** [comment]
@anijain2305 @jansel why do we have guards for `list, tuple` literals, but not sets?
https://github.com/pytorch/pytorch/blob/c536903c3fd125ddc901ffc4a3fa26aa96f446b9/torch/_dynamo/variables/constant.py#L47-L57

File: torch/_dynamo/variables/constant.py [hunk 1]
```diff
@@ -38,18 +38,14 @@ class ConstantVariable(VariableTracker):
...
-        if not is_literal:
-            for disallowed_type, reason in _type_to_assert_reason.items():
-                assert not isinstance(value, disallowed_type), reason
 
-        # Routing for list and tuple literals.
-        if is_literal and isinstance(value, (set, frozenset)):
+        # Routing for supported collection literals.
+        if isinstance(value, (set, frozenset)):
             items = []
             for i, x in enumerate(value):
```
Comment (jansel) [inline torch/_dynamo/variables/constant.py:45]:
```suggestion
            for x in value:
```

**Review by jansel — APPROVED** [review]

**Comment by jansel** [comment]
> @anijain2305 @jansel why do we have guards for `list, tuple` literals, but not sets?
> 
> https://github.com/pytorch/pytorch/blob/c536903c3fd125ddc901ffc4a3fa26aa96f446b9/torch/_dynamo/variables/constant.py#L47-L57

This might be a missing guard?  If so we should add it.  You could write a test where this changes and see if we recompile properly.

**Comment by StrongerXi** [comment]
> This might be a missing guard? If so we should add it. You could write a test where this changes and see if we recompile properly.

Cool, I'll give it a test and send a follow-up patch if needed.

**Pull request closed without merging**